spec/models/user_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validations' do
+    it 'name must be present' do
+      user = User.new(name: '')
+
+      user.valid?
+
+      expect(user.errors[:name]).to include("can't be blank")
+    end
+
+    it 'post_counter must be an integer' do
+      user = User.new(post_counter: 'a')
+
+      user.valid?
+
+      expect(user.errors[:post_counter]).to include('is not a number')
+    end
+
+    it 'post_counter must be greater than or equal to 0' do
+      user = User.new(post_counter: -1)
+
+      user.valid?
+
+      expect(user.errors[:post_counter]).to include('must be greater than or equal to 0')
+    end
+
+    it 'returns three recent posts' do
+      user = User.create(name: 'otsama', photo: 'photo', bio: 'software developer', post_counter: 5)
+      post4 = Post.create(title: 'test title 4', text: 'test text', comments_counter: 1, likes_counter: 1,
+                          author_id: user.id)
+      post5 = Post.create(title: 'test title 5', text: 'test text', comments_counter: 1, likes_counter: 1,
+                          author_id: user.id)
+      post6 = Post.create(title: 'test title 6', text: 'test text', comments_counter: 1, likes_counter: 1,
+                          author_id: user.id)
+
+      expect(user.recent_posts).to eq([post6, post5, post4])
+    end
```

Comment:
- [ ] Great job working on the validations so far but I think it would be great if you fix the failing test since there is one failing tests in user model after running them I faced this error message so kindly take a look at it and try to solve them 🤩🚀💪🏻


- I think because of your author_id is not named like this in your DB it is user_id so you should name it the same so your tests will passed ✨🤩✅✅


![image](https://github.com/Othniel2471/blog_app/assets/58553711/c07157b2-b189-4e13-97e6-5fd974960d20)

Reply:
Hello @Othniel2471 I think its name `user_id` I checked the schema well and I am sure that I switched to the branch `model-validation` but I am not sure what is the problem exactly!


![image](https://github.com/Othniel2471/blog_app/assets/58553711/14e2d165-7b74-4743-9f9f-2c21b647f791)